aiida/orm/nodes/data/remote/base.py

```diff
@@ -149,18 +149,22 @@ def listdir_withattributes(self, path='.'):
                 else:
                     raise
 
-    def _clean(self):
+    def _clean(self, transport=None):
         """
         Remove all content of the remote folder on the remote computer
         """
         from aiida.orm.utils.remote import clean_remote
 
-        authinfo = self.get_authinfo()
-        transport = authinfo.get_transport()
         remote_dir = self.get_remote_path()
+        if transport is None:
+            authinfo = self.get_authinfo()
+            transport = authinfo.get_transport()
 
-        with transport:
+            with transport:
+                clean_remote(transport, remote_dir)
+        else:
```

Comment:
In this clause we should really check that the `transport` goes with the `Computer` of the `remote_dir`. There is no direct way of getting the `Computer` from `transport`, but I think we should be able to get the `hostname` and then we compare that with the `hostname` of the `Computer`. Would that be sufficient?

Reply:
The hostname from the transport is stored as `Transport._machine`, but this is something defined by the plugins. I compare this to `Computer.hostname`. Hope this is OK?